src/Protocol/Diaspora/Repository/DiasporaContact.php

```diff
@@ -0,0 +1,260 @@
+	{
+		return $this->selectOne(['addr' => $uri->getAddr()]);
+	}
+
+	/**
+	 * @param int $uriId
+	 * @return bool
+	 * @throws \Exception
+	 */
+	public function existsByUriId(int $uriId): bool
+	{
+		return $this->db->exists(self::$table_name, ['uri-id' => $uriId]);
+	}
+
+	public function save(Entity\DiasporaContact $DiasporaContact): Entity\DiasporaContact
+	{
+		$uriId = $DiasporaContact->uriId ?? ItemURI::insert(['uri' => $DiasporaContact->url, 'guid' => $DiasporaContact->guid]);
+
+		$fields = [
+			'uri-id'            => $uriId,
+			'addr'              => $DiasporaContact->addr,
+			'alias'             => (string)$DiasporaContact->alias,
+			'nick'              => $DiasporaContact->nick,
+			'name'              => $DiasporaContact->name,
+			'given-name'        => $DiasporaContact->givenName,
+			'family-name'       => $DiasporaContact->familyName,
+			'photo'             => (string)$DiasporaContact->photo,
+			'photo-medium'      => (string)$DiasporaContact->photoMedium,
+			'photo-small'       => (string)$DiasporaContact->photoSmall,
+			'batch'             => (string)$DiasporaContact->batch,
+			'notify'            => (string)$DiasporaContact->notify,
+			'poll'              => (string)$DiasporaContact->poll,
+			'subscribe'         => (string)$DiasporaContact->subscribe,
+			'searchable'        => $DiasporaContact->searchable,
+			'pubKey'            => $DiasporaContact->pubKey,
+			'baseurl'           => (string)$DiasporaContact->baseurl,
+			'gsid'              => $DiasporaContact->gsid,
+			'updated'           => DateTimeFormat::utcNow(),
+			'interacting_count' => $DiasporaContact->interacting_count,
+			'interacted_count'  => $DiasporaContact->interacted_count,
+			'post_count'        => $DiasporaContact->post_count,
+		];
+
+		// Limit the length on incoming fields
+		$fields = $this->definition->truncateFieldsForTable('diaspora-contact', $fields);
+
+		$this->db->insert('diaspora-contact', $fields, Database::INSERT_UPDATE);
+
+		return $this->selectOneByUriId($uriId);
+	}
+
+	/**
+	 * Fetches a Diaspora profile from a given URI and updates it depending on the mode
+	 *
+	 * @param WebFingerUri $uri    profile URI
+	 * @param boolean      $update true = always update, false = never update, null = update when not found or outdated
+	 * @return Entity\DiasporaContact
+	 * @throws HTTPException\NotFoundException
+	 */
+	public function getByAddr(WebFingerUri $uri, ?bool $update = self::UPDATE_IF_MISSING_OR_OUTDATED): Entity\DiasporaContact
+	{
+		if ($update !== self::ALWAYS_UPDATE) {
+			try {
+				$dcontact = $this->selectOneByAddr($uri);
+				if ($update === self::NEVER_UPDATE) {
+					return $dcontact;
+				}
+			} catch (HTTPException\NotFoundException $e) {
+				if ($update === self::NEVER_UPDATE) {
+					throw $e;
+				}
+			}
+		}
+
+		// Trigger a contact update if needed
+		$contact = Contact::getByURL($uri, $update, ['uri-id', 'network', 'next-update']);
```

Comment:
Do you need all these fields here?

Reply:
Yes, they're used inside `Contact::getByURL` to determine if the contact needs to be updated 🤷‍♂️

Reply:
Okay. I wasn't sure if this was some kind of leftover.

Reply:
Never mind, these fields were already automatically added inside of `Contact::getByURL`.